### PR TITLE
dts: arm: st: add support for STM32F205xx SOC

### PR DESCRIPTION
**Pull request by Mforcen**
Signed-off-by: Manuel Forcén Muñoz <manuforcen@gmail.com>

STM32F205xx device tree include files added.
Tested properly in a custom board.

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]
Thanks for this addition. Please, make clock fix part of one commit and new SoC addition part of another one. Commit message can also be improved, e.g. `dts: arm: st: add support for xxx`. https://docs.zephyrproject.org/latest/contribute/index.html

File: dts/arm/st/f2/stm32f2.dtsi [hunk 1]
```diff
@@ -172,7 +172,7 @@
 		usart1: serial@40011000 {
 			compatible = "st,stm32-usart", "st,stm32-uart";
 			reg = <0x40011000 0x400>;
-			clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00004000>;
```
[the remark this answers is not in the document]
Comment (Mforcen) [reply dts/arm/st/f2/stm32f2.dtsi:175]: This was changed in the latedt commit

File: dts/arm/st/f2/stm32f2.dtsi [hunk 2]
```diff
@@ -172,7 +172,7 @@
...
+			clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00000010>;
```
Comment (erwango) [inline dts/arm/st/f2/stm32f2.dtsi:175]: Already fixed in https://github.com/zephyrproject-rtos/zephyr/pull/32028 based on the issue you reported.
Commit could be removed.

**Review by erwango — CHANGES_REQUESTED** [review]

File: dts/arm/st/f2/stm32f2.dtsi (hunk 2 above)
Comment (Mforcen) [reply dts/arm/st/f2/stm32f2.dtsi:175]: I'm sorry, I'm kind of new to all these things about pull requests.
I'll try my best.
I'm going to remove the clock commit.

**Review by erwango — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**